go/test/endtoend/cluster/cluster_process.go

```diff
@@ -70,17 +78,40 @@ type Shard struct {
 	Vttablets []Vttablet
 }
 
+// MasterTablet get the 1st tablet which is master
+func (shard *Shard) MasterTablet() *Vttablet {
+	return &shard.Vttablets[0]
+}
+
+// Rdonly get the last tablet which is rdonly
+func (shard *Shard) Rdonly() *Vttablet {
+	if len(shard.Vttablets) > 2 {
+		return &shard.Vttablets[len(shard.Vttablets)-1]
+	}
+	return nil
+}
+
+// Replica get the last but one tablet which is replica
+// Mostly we have either 2 or 3 tablet setup [master, replica], [master, replica, rdonly]
```

Comment:
this won't work in a 2 tablet setup?

Reply:
I will change the comment, as mostly we test with 3 tablets or more.